pkg/sdn/registry/netnamespace/migration.go

```diff
@@ -0,0 +1,105 @@
+package netnamespace
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+
+	"k8s.io/kubernetes/pkg/api/errors"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/storage"
+	kerrors "k8s.io/kubernetes/pkg/util/errors"
+
+	"github.com/openshift/openshift-sdn/pkg/netid"
+	"github.com/openshift/origin/pkg/sdn/api"
+)
+
+const (
+	etcdNetNamespacePrefix string = "/registry/sdnnetnamespaces"
+)
+
+type Migrate struct {
+	etcdHelper storage.Interface
+	client     kclient.NamespaceInterface
+}
+
+func NewMigrate(etcdHelper storage.Interface, client kclient.NamespaceInterface) *Migrate {
+	return &Migrate{
+		etcdHelper: etcdHelper,
+		client:     client,
+	}
+}
+
+func (m *Migrate) Run() error {
+	var netnsList api.NetNamespaceList
+
+	err := m.etcdHelper.List(context.TODO(), etcdNetNamespacePrefix, "", storage.Everything, &netnsList)
+	if err != nil {
+		return fmt.Errorf("List network namespaces failed: %v, please retry the operation", err)
+	}
+
+	errList := []error{}
+	for _, netns := range netnsList.Items {
+		if err := m.migrateNetNamespace(&netns); err != nil {
+			errList = append(errList, err)
+			continue
+		}
+	}
+
+	if len(errList) > 0 {
+		errList = append(errList, fmt.Errorf("Unexpected errors during NetNamespace migration, please retry the operation"))
+	}
+	return kerrors.NewAggregate(errList)
+}
+
+func (m *Migrate) deleteNetNamespace(netns *api.NetNamespace) {
+	var out api.NetNamespace
+	if err := m.etcdHelper.Delete(context.TODO(), path.Join(etcdNetNamespacePrefix, netns.ObjectMeta.Name), &out, nil); err != nil {
+		glog.Errorf("Failed to delete NetNamespace: %v, error: %v", netns, err)
+	}
+}
+
+func (m *Migrate) migrateNetNamespace(netns *api.NetNamespace) error {
+	// Get corresponding namespace for the NetNamespace object
+	ns, err := m.client.Get(netns.NetName)
+	// If corresponding namespace no longer exists, nothing to migrate in this case.
+	if errors.IsNotFound(err) {
+		m.deleteNetNamespace(netns)
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Get namespace %q failed: %v", netns.NetName, err)
+	}
+
+	// Set netid as namespace annotation
+	if err := netid.SetVNID(ns, netns.NetID); err != nil {
+		return fmt.Errorf("Set netid %d for namespace %q failed: %v", netns.NetID, ns.Name, err)
+	}
+
+	// Update namespace
+	updatedNs, err := m.client.Update(ns)
+	if errors.IsNotFound(err) {
+		m.deleteNetNamespace(netns)
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Update namespace %q failed: %v", ns.Name, err)
+	}
+
+	// Validate
+	if id, err := netid.GetVNID(updatedNs); err == nil {
+		if id == netns.NetID {
```

Comment:
invert this condition and return early.  Always return early to avoid creating an else.